Java/src/main/java/com/gildedrose/GildedRose.java

```diff
@@ -8,55 +8,81 @@ public GildedRose(Item[] items) {
     }
 
     public void updateQuality() {
+
         for (int i = 0; i < items.length; i++) {
-            if (!items[i].name.equals("Aged Brie")
-                    && !items[i].name.equals("Backstage passes to a TAFKAL80ETC concert")) {
-                if (items[i].quality > 0) {
-                    if (!items[i].name.equals("Sulfuras, Hand of Ragnaros")) {
-                        items[i].quality = items[i].quality - 1;
-                    }
-                }
-            } else {
-                if (items[i].quality < 50) {
-                    items[i].quality = items[i].quality + 1;
-
-                    if (items[i].name.equals("Backstage passes to a TAFKAL80ETC concert")) {
-                        if (items[i].sellIn < 11) {
-                            if (items[i].quality < 50) {
-                                items[i].quality = items[i].quality + 1;
-                            }
-                        }
-
-                        if (items[i].sellIn < 6) {
-                            if (items[i].quality < 50) {
-                                items[i].quality = items[i].quality + 1;
-                            }
-                        }
-                    }
-                }
+
+            switch (items[i].name) {
+                case "Aged Brie":
+                    updateBrie(items[i]);
+                    updateSellin(items[i]);
+                    break;
+
+                case "Backstage passes to a TAFKAL80ETC concert":
+                    updateBackstagePasses(items[i]);
+                    updateSellin(items[i]);
+                    break;
+
+                case "Sulfuras, Hand of Ragnaros":
+                    break;
+            
+                default:
+                    updateCommonItem(items[i]);
+                    updateSellin(items[i]);
+                    break;
+            }
+            
+        }
+    }
+
+    public void updateSellin(Item Item) {
+        Item.sellIn = Item.sellIn - 1;
+    }
+
+    public void updateCommonItem(Item commonItem) {
+
+        if (commonItem.quality > 0) {
+            commonItem.quality -= 1;
+        }
+
+        if (commonItem.sellIn < 0) { 
+            if (commonItem.quality > 0) {
+                commonItem.quality -= 1;
             }
+        }
+
+    }
+
+    public void updateBrie(Item brie) {
 
-            if (!items[i].name.equals("Sulfuras, Hand of Ragnaros")) {
-                items[i].sellIn = items[i].sellIn - 1;
+        if(brie.quality < 50) {
+            brie.quality += 1;
+        }
+
+        if (brie.sellIn <= 0 && brie.quality < 50) {
+            brie.quality += 1;
+        }
+    }
+
+    public void updateBackstagePasses(Item backstagePass) {
+
+        if(backstagePass.quality < 50) {
```

Comment:
why do you have the "if(backstagePass.quality < 50)" so many times

Reply:
in general if (item.quality < 50 ) increase could be moved to a method